### ISSUE TITLE
Callbacks only detected when they are added to rust first then slint

### ISSUE THREAD

**Issue by Oevin7** [slint-ui/slint#4840]
I am on Zorin OS and I am using rust with slint on VS Codium.

So when I add a new callback to slint first and then write it in rust, rust throws an error saying the callback does not exsist in the struct. However when I do it the other way around and write it in rust and then slint it works. 

so if I write: callback reset-counter(); 
and then I go to write the functionality in rust say: ui.on_reset_counter(move || {});

rust throws the error. 

If I then remove callback reset-counter(); and then paste it back into the slint file rust will fix the error. I'm not sure this is intended and it did throw me off a bit as I am trying to learn slint to make UI apps in rust.

**Comment by ogoffart** [comment]
Thanks for reporting a bug.
What error exactly does it throws?
Could you paste the complete source code as well as the error?

**Comment by Oevin7** [comment]
> Thanks for reporting a bug. What error exactly does it throws? Could you paste the complete source code as well as the error?

This is the error I recieve: no method named `on_remove_text` found for struct `AppWindow` in the current scope
method not found in `AppWindow`

After you create the callback it doesn't matter if it is deleted from slint and rust. It will work in whatever order, but on your first attemt to create the method it has to be done in rust first. I tried recreating it and it wouldn't work for my already created callbacks, so I had to make a new one and the same error appeared.

To get it to work I have to paste the callback and move it to a different line. I am new to slint so and rust so I could be doing something wrong, but the behavior does seem a little odd to me.

Thank you for the quick response!

**Comment by ogoffart** [comment]
Could you please post a screenshot so i can see exactly what error that is?

**Comment by Oevin7** [comment]
Hey! I'm so sorry! I've been really busy the past few months and haven't had time to respond. I also never got a notification for this. I did just realize now I may have been dumb lol. I am very sorry fore the trouble! I was using VsCodium and I am not used to having to save beforehand as I usually use a JetBrains IDE. Therefore, I had to save the Rust file before I could access the new callback. It was my lack of understanding rather than a bug! I'm sorry for all the trouble and being a pain. Anywho you guys have a cool project and I can't wait to keep learning how to use it :)
